### ISSUE TITLE
mismatch in ttnn. softmax_in_place

### ISSUE THREAD

**Issue by ranjith-m-s** [tenstorrent/tt-metal#36881]
### Component / Area

ttnn

### Issue Type (optional)

Bad Outputs

### Observed

the softmax for the specified shapes with the given program config run at expected 6 us on n300 (using 1 chip), but the output doesn't match the reference torch version.

### Expected

matches the torch ref

### 1. Steps (exact commands)

`import pytest
import torch
import ttnn

from models.common.utility_functions import (
    tt2torch_tensor,
    comp_pcc,
)

@pytest.mark.parametrize("dtype", [ ttnn.bfloat16])
def test_sotmax(device, dtype):
    seq_len = 1024
    torch.manual_seed(42)

    # Inputs
    scores = torch.randn((2, 16, seq_len, seq_len), dtype=torch.bfloat16)
    # key = torch.randn((1, 1, 64, seq_len), dtype=torch.bfloat16)

    program_config = ttnn.SoftmaxShardedMultiCoreProgramConfig(
                    compute_with_storage_grid_size=(8, 8),
                    subblock_w=1,
                    block_h=1,
                    block_w=1,
                )

    in0_memory_config = ttnn.create_sharded_memory_config(
                    scores.shape,  # tensor shape
                    core_grid=ttnn.CoreGrid(y=8, x=8),
                    # core_grid=ttnn.CoreGrid(y=8, x=1),
                    strategy=ttnn.ShardStrategy.HEIGHT,
                    orientation=ttnn.ShardOrientation.ROW_MAJOR,
                )

    scores_tt = ttnn.from_torch(
                    scores,
                    tile=ttnn.Tile((32, 32)),
                    dtype=dtype,
                    layout=ttnn.TILE_LAYOUT,
                    device=device,
                    # memory_config=in0_memory_config,
                    # memory_config=ttnn.L1_MEMORY_CONFIG,
                )

    scores_tt = ttnn.to_memory_config(scores_tt, in0_memory_config)
    
    compute_config = ttnn.WormholeComputeKernelConfig(
            # math_fidelity=ttnn.MathFidelity.LoFi,
            math_fidelity=ttnn.MathFidelity.HiFi4,
            fp32_dest_acc_en=True,
        )
    scores_tt = ttnn.softmax_in_place(scores_tt,
                    program_config=program_config,
                    compute_kernel_config=compute_config,
                    numeric_stable = True,
                )
    
    tt_out_torch = tt2torch_tensor(scores_tt)
    does_pass, pcc_msg = comp_pcc(torch.nn.functional.softmax(scores), tt_out_torch, 0.98)
    print(pcc_msg)
    assert does_pass, "❌ softmax output mismatch!"

    print("✅ softmax TTNN runs sucessfully")`

### 2. Input data / link or description

-

### 3. Frequency

always

### 1. Software Versions

Ubuntu 22.04.5 LTS
commit https://github.com/tenstorrent/tt-metal/commit/7a79eba540178455e32c34189750deebc455dbd0

### 2. Hardware Details

n300

### Is this a regression?

Unknown

### Priority

P0

**Comment by fahmedchanTT** [comment]
@bbradelTT Please take a look .
